lib/Jsrt/ChakraDebug.h

```diff
@@ -0,0 +1,567 @@
+            _In_ JsDiagStepType stepType);
+
+    /// <summary>
+    ///     Gets list of scripts.
+    /// </summary>
+    /// <param name="scriptsArray">Array of script objects.</param>
+    /// <remarks>
+    ///     <para>
+    ///     [{
+    ///         "scriptId" : 2,
+    ///         "fileName" : "c:\\Test\\Test.js",
+    ///         "lineCount" : 4,
+    ///         "sourceLength" : 111
+    ///       }, {
+    ///         "scriptId" : 3,
+    ///         "parentScriptId" : 2,
+    ///         "scriptType" : "eval code",
+    ///         "lineCount" : 1,
+    ///         "sourceLength" : 12
+    ///     }]
+    ///     </para>
+    /// </remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The current runtime should be in debug state. This API can be called when runtime is at a break or running.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagGetScripts(
+            _Out_ JsValueRef *scriptsArray);
+
+    /// <summary>
+    ///     Gets source for a specific script identified by scriptId from JsDiagGetScripts.
+    /// </summary>
+    /// <param name="scriptId">Id of the script.</param>
+    /// <param name="source">Source object.</param>
+    /// <remarks>
+    ///     <para>
+    ///     {
+    ///         "scriptId" : 1,
+    ///         "fileName" : "c:\\Test\\Test.js",
+    ///         "lineCount" : 12,
+    ///         "sourceLength" : 15154,
+    ///         "source" : "var x = 1;"
+    ///     }
+    ///     </para>
+    /// </remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The current runtime should be in debug state. This API can be called when runtime is at a break or running.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagGetSource(
+            _In_ unsigned int scriptId,
+            _Out_ JsValueRef *source);
+
+    /// <summary>
+    ///     Gets the source information for a function object.
+    /// </summary>
+    /// <param name="function">JavaScript function.</param>
+    /// <param name="functionInfo">Function info, scriptId, start line, start column, line number of first statement, column number of first statement.</param>
+    /// <remarks>
+    ///     <para>
+    ///     {
+    ///         "scriptId" : 1,
+    ///         "fileName" : "c:\\Test\\Test.js",
+    ///         "line" : 1,
+    ///         "column" : 2,
+    ///         "firstStatementLine" : 6,
+    ///         "firstStatementColumn" : 0
+    ///     }
+    ///     </para>
+    /// </remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     This API can be called when runtime is at a break or running.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagGetFunctionPosition(
```

Comment:
position is more appropriate as this is position in source info, should I change functionInfo to functionPosition?